### PR TITLE
git-terminal.md: add instructions for CRLF on Windows

### PR DESCRIPTION
**Pull request by trolzen**
These are instructions to avoid some strange warnings when editing pages in editorconfig-aware editor on Windows.

See #7097

### CODE REVIEW

**Review by kbdharun — APPROVED** [review]
This is essential documentation. Thanks for adding it.

**Comment by navarroaxel** [comment]
@bl-ue could you please help here?

**Comment by trolzen** [comment]
Is  there anything I can do to help this PR be merged?

File: contributing-guides/git-terminal.md [hunk 1]
```diff
@@ -90,3 +90,27 @@ git push --force-with-lease
...
+warning: LF will be replaced by CRLF in...
+```
+
+To handle this problem, you need to clone the repository using the command:
+
+```sh
+git clone --config core.eol=lf <repository>
+```
+
+If you've already cloned the repository, and don't want to repeat the whole process (if, for example, you've already made some modifications), you can fix it using the following commands. Be careful as those commands are potentially dangerous and you can lose your unfinished work in the current repository!
```
Comment (waldyrious) [inline contributing-guides/git-terminal.md:108]: Just a small adjustment:

```suggestion
If you've already cloned the repository, and don't want to repeat the whole process (if, for example, you've already made some modifications), you can fix the issue using the following commands. Be careful as these commands are potentially dangerous and you can lose your unfinished work in the current repository!
```

File: contributing-guides/git-terminal.md [hunk 2]
```diff
@@ -90,3 +90,27 @@ git push --force-with-lease
...
+
+```sh
+git config --local core.eol lf
+git stash push
+git rm -rfq --cached .
+git reset --hard HEAD
+git stash pop --index
```
Comment (waldyrious) [inline contributing-guides/git-terminal.md:115]: Hmm, would you mind adding comments next to each command explaining roughly what they do? I don't like the idea of recommending users to run commands (especially potentially dangerous ones, as pointed above) without them understanding what they are doing.

Comment (trolzen) [inline contributing-guides/git-terminal.md:115]: Added some comments. What do you think?
```suggestion
git config --local core.eol lf  # set the correct option value
git stash push                  # stash any local changes (if working tree is clean, skip this and the last one)
git rm -rfq --cached .          # remove all the files under version control from index
git reset --hard HEAD           # update all files in index and working tree without converting LF to CRLF, as per new option value
git stash pop --index           # restore the previous state from stash
```

Comment (trolzen) [reply contributing-guides/git-terminal.md:115]: Added some comments. What do you think?
```suggestion
git config --local core.eol lf  # set the correct option value
git stash push                  # stash any local changes (if working tree is clean, skip this and the last one)
git rm -rfq --cached .          # remove all the files under version control from index
git reset --hard HEAD           # update all files in index and working tree without converting LF to CRLF, as per new option value
git stash pop --index           # restore the previous state from stash
```

File: contributing-guides/git-terminal.md [hunk 3]
```diff
@@ -90,3 +90,27 @@ git push --force-with-lease
...
+There is an issue that could arise when you clone the repository under Windows and use an editor which honors the settings in the `.editorconfig` file. With the default configuration, when you initially clone the repository, Git checks out files converting line endings to `CRLF`. Later, when you edit some file, or just save it without any modifications, your editor converts line endings to `LF` as per configuration in the `.editorconfig`. This causes the confusion, making Git mark the files as modified whereas they are not and issue the following warnings on `git diff` and `git add`:
+
+```
+warning: LF will be replaced by CRLF in...
+```
+
+To handle this problem, you need to clone the repository using the command:
+
+```sh
+git clone --config core.eol=lf <repository>
```
Comment (trolzen) [inline contributing-guides/git-terminal.md:105]:
```suggestion
git clone --config core.eol=lf {{repository}}
```
I suppose it is better to use the notation adopted in this project.

Comment (navarroaxel) [reply contributing-guides/git-terminal.md:105]: Based on the `git-clone` page
```suggestion
git clone --config core.eol=lf {{remote_repository_location}}
```

File: contributing-guides/git-terminal.md (hunk 2 above)
Comment (waldyrious) [reply contributing-guides/git-terminal.md:115]: Looks good! The only changes I'd make are "if working tree" → "if the working tree", and "skip this" → "skip this step". Other than that, please go ahead and add these :)

Comment (navarroaxel) [reply contributing-guides/git-terminal.md:115]: Perhaps this better than a lot of spaces
```suggestion
# set line feed (LF) as end of line
git config --local core.eol lf
# stash any local changes (if working tree is clean, skip this and the last one)
git stash push
# remove all the files under version control from index
git rm -rfq --cached .
# update all files in index and working tree without converting LF to CRLF, as per new option value
git reset --hard HEAD
# restore the previous state from stash
git stash pop --index
```

Comment (trolzen) [reply contributing-guides/git-terminal.md:115]:
```suggestion
# set line feed (LF) as end of line
git config --local core.eol lf
# stash any local changes (if the working tree is clean, skip this step and the last one)
git stash push
# remove all the files under version control from index
git rm -rfq --cached .
# update all the files in index and working tree without converting LF to CRLF, as per new option value
git reset --hard HEAD
# restore the previous state from stash
git stash pop --index
```

**Comment by trolzen** [comment]
@waldyrious @navarroaxel I've updeted the PR with all the suggestions.

**Comment by trolzen** [comment]
Why this one check doesn't pass?
```
CI Expected — Waiting for status to be reported
```

**Comment by kbdharun** [comment]
> Why this one check doesn't pass?
> 
> ```
> CI Expected — Waiting for status to be reported
> ```

That is a known issue with GitHub API, closing and immediately reopening the PR will fix it (**Don't delete the PRs branch**)

**Review by navarroaxel — APPROVED** [review]

**Pull request merged**